lib/chip-io.js

```diff
@@ -253,42 +242,24 @@ ChipIO.prototype.i2cReadOnce = function(address, register, size, handler) {
 
   var bus = typeof(addressToBus[address]) !== 'undefined' ? addressToBus[address] : 2;
   var i2c = new I2C(bus, address);
+  var event;
+  var data;
 
-  i2c.open(function(err) {
-    if (err) {
-      return;
-    }
-
-    if (register) {
-      i2c.readRegister(register, size, function(err, data) {
-        if (err) {
-          return;
-        }
-
-        i2c.close(function() {
-          var event = 'I2C-reply-' + address + '-' + register;
-
-          this.once(event, handler);
+  i2c.open();
 
-          this.emit(event, data);
-        }.bind(this));
-      }.bind(this));
-    } else {
-      i2c.read(size, function(err, data) {
-        if (err) {
-          return;
-        }
+  if (register) {
+    event = 'I2C-reply-' + address + '-' + register;
+    data = i2c.readRegister(register, size);
+  } else {
+    event = 'I2C-reply-' + address;
+    data = i2c.read(size);
```

Comment:
I've switched it to use `process.nextTick` for now. We can look at it again when we switch to use `i2s-bus`.